### ISSUE TITLE
WithEvents and New keyword

### ISSUE THREAD

**Issue by Kr00l** [WaynePhillipsEA/twinbasic#357]
**Describe the bug**
In VBx it is not possible auto-declare New on a object that has the WithEvents keyword.
![image](https://user-images.githubusercontent.com/33756666/129522659-6b202411-fc49-4e8b-866a-253745c6402c.png)

**To Reproduce**
```
Class MyEventClass
    Public Event MyEvent()
    
    Public Sub RaiseMyEvent()
        RaiseEvent MyEvent()
    End Sub
End Class

Class MyClass
    Private WithEvents MyEvents As New MyEventClass
    
    Private Sub Class_Initialize()
        'Set MyEvents = New MyEventClass
    End Sub
    
    Private Sub MyEvents_MyEvent()
        MsgBox "Hello World"
    End Sub
    
    Public Sub Test()
        MyEvents.RaiseMyEvent
    End Sub
    
End Class

Module MainModule

    ' This project type is set to 'Standard EXE' in the Settings file, so you need a Main() subroutine to run when the EXE is started.
    Public Sub Main()
        Dim This As MyClass = New MyClass
        This.Test
    End Sub
	
End Module
```

**Expected behavior**
in tB this gives not a compile error. Instead it gives a runtime error.
![image](https://user-images.githubusercontent.com/33756666/129522796-0a23f787-a89b-4d98-941b-b9fea6ebe51f.png)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - twinBASIC compiler version [e.g. 0.10.4386]

**Additional context**
Doing the alternative tB style declare
`Private WithEvents MyEvents As MyEventClass = New MyEventClass`
crashes the compiler.

So, two options.
1. tB should throw a compile error like in VBx instead of a runtime error and only allows when explicitly set `Set MyEvents = New MyEventClass`
2. tB could be better and allow it. Don't know if it is feasible or not and if it has side-effects or not.

**Comment by WaynePhillipsEA** [comment]
I think we should go for #1 and behave like VBx here.

If we were to allow As-New semantics here, the events won't bind until the WithEvents field is first accessed/touched, and that seems a bit weird to me.  By having the WithEvrnts field manually instantiated, it is clear as to when the events are subscribed to.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.4503.  Thanks for reporting!

(the field initializer version should now be working also: `Private WithEvents MyEvents As MyEventClass = New MyEventClass`)
